src/gui/sidebar/previews/layer/SidebarPreviewLayerEntry.cpp

```diff
@@ -6,13 +6,29 @@
 #include "i18n.h"
 
 
-SidebarPreviewLayerEntry::SidebarPreviewLayerEntry(SidebarPreviewBase* sidebar, const PageRef& page, int layer,
+SidebarPreviewLayerEntry::SidebarPreviewLayerEntry(SidebarPreviewLayers* sidebar, const PageRef& page, int layer,
                                                    const std::string& layerName, size_t index, bool stacked):
         SidebarPreviewBaseEntry(sidebar, page),
         index(index),
         layer(layer),
+        sidebar(sidebar),
         stacked(stacked),
         box(gtk_box_new(GTK_ORIENTATION_VERTICAL, 2)) {
+
+    const auto clickCallback = G_CALLBACK(+[](GtkWidget* widget, GdkEvent* event, SidebarPreviewLayerEntry* self) {
+        // Open context menu on right mouse click
+        if (event->type == GDK_BUTTON_PRESS) {
+            auto mouseEvent = reinterpret_cast<GdkEventButton*>(event);
+            if (mouseEvent->button == 3) {
+                self->mouseButtonPressCallback();
+                self->sidebar->openPreviewContextMenu();
+                return true;
+            }
+        }
+        return false;
+    });
+    g_signal_connect_after(this->widget, "button-press-event", clickCallback, this);
```

Comment:
Because we now have `clickCallback`s in both `SidebarPreviewLayerEntry` and `SidebarPreviewPageEntry`, I think it might make sense to factor this code out into `SidebarPreviewBaseEntry`.

`void openPreviewContextMenu(SidebarPreviewBaseEntry&)` could be a `virtual` method of `SidebarPreviewBase` that we `override` in `SidebarPreviewLayer` and `SidebarPreviewPage`.

This design would be particularly useful if we want to make long taps open the context menu in the future (on systems like Linux/Wayland).

Reply:
> [abstract `clickCallback`]

Sure. To do that as far as I can tell we'll need to either
- let `SidebarPreviewBaseEntry` be a `friend` of `SidebarPreviewBase` (I'll do this for now)
- make `SidebarPreviewBase::openPreviewContextMenu()` public